analyses/DESeq_analysis/run-DESeq-analysis.R

```diff
@@ -0,0 +1,195 @@
+Cancer_Histology <- paste("Combined",Cancer_Histology[which(!is.na(Cancer_Histology))],sep="_")
+
+
+#Save all the GTEx subgroups represented in the countsdata into a variable. Remove all 'NA's 
+Gtex_Tissue_subgroup = Gtex_Tissue_subgroup[!is.na(Gtex_Tissue_subgroup)]
+
+#Create an empty df to populate with rbind of all normal Kids_First_Biospecimen_ID and gtex_subgroup
+#Create DF that list all Kids_First_Biospecimen_IDs by GTEX subgroup
+sample_type_df_normal = data.frame()
+for(I in 1:length(Gtex_Tissue_subgroup))
+{
+  sample_type_df_normal = rbind(sample_type_df_normal,data.frame(Case_ID = hist.filtered$Kids_First_Biospecimen_ID[which(hist.filtered$gtex_subgroup == Gtex_Tissue_subgroup[I])],Type = Gtex_Tissue_subgroup[I]))
+}
+
+#Create an empty df to populate with rbind of all tumor Kids_First_Biospecimen_ID and cancer_group
+#Create DF that list all Kids_First_Biospecimen_IDs by cancer group subgroup
+sample_type_df_tumor = data.frame()
+for(I in 1:length(Cancer_Histology))
+{
+  sample_type_df_tumor = rbind(sample_type_df_tumor,data.frame(Case_ID = hist.filtered$Kids_First_Biospecimen_ID[which(hist.filtered$cancer_group == gsub("Combined_","",Cancer_Histology[I]))],Type=Cancer_Histology[I]))
+}
+
+#Create an empty df to populate with rbind of all tumor Kids_First_Biospecimen_ID and cancer_group by cohort
+#Create DF that list all Kids_First_Biospecimen_IDs by Cohort - Cancer groups
+sample_type_df_tumor_cohort = data.frame()
+for(I in 1:length(Cancer_Histology_COHORT))
+{
+  Cancer_Histology_COHORT_cohort =  strsplit(Cancer_Histology_COHORT[I],split="_")[[1]][1]
+  Cancer_Histology_COHORT_cancer_group =  strsplit(Cancer_Histology_COHORT[I],split="_")[[1]][2]
+  sample_type_df_tumor_cohort = rbind(sample_type_df_tumor_cohort,data.frame(Case_ID = hist.filtered$Kids_First_Biospecimen_ID[which(hist.filtered$cancer_group == Cancer_Histology_COHORT_cancer_group & hist.filtered$cohort == Cancer_Histology_COHORT_cohort)],Type=Cancer_Histology_COHORT[I]))
+}
+
+#Combine the rows from the normal and tumor sample df
+sample_type_df = rbind(sample_type_df_tumor,sample_type_df_tumor_cohort,sample_type_df_normal)
+
+#Filter one more to ensure the rownames in the countsdata file match the sample dataframe for DEG just created
+countData_filtered_DEG = countData_filtered[,which(colnames(countData_filtered) %in% sample_type_df$Case_ID)]
+
+#Filter one more to ensure the rownames in the countsdata file match the sample dataframe for DEG just created
+sample_type_df_filtered = unique(sample_type_df[which(sample_type_df$Case_ID %in% colnames(countData_filtered_DEG)),])
+
+#Define All cancer groups (Combined and cohort-specific) in the histology list
+histology_filtered = unique(sample_type_df_filtered$Type[-grep("GTEX",sample_type_df_filtered$Case_ID)])
+
+#Define All GTEx groups as normal in the GTEX_filtered list
+GTEX_filtered = unique(sample_type_df_filtered$Type[grep("GTEX",sample_type_df_filtered$Case_ID)])
```

Comment:
Some ID strings may have random "GTEX" in them, so add `"^"` to ensure match from beginning.

It might also be informative to note somewhere in the module that all GTEx sample Kids_First_Biospecimen_IDs are assumed to start with "GTEX".

```suggestion
histology_filtered = unique(sample_type_df_filtered$Type[-grep("^GTEX",sample_type_df_filtered$Case_ID)])

#Define All GTEx groups as normal in the GTEX_filtered list
GTEX_filtered = unique(sample_type_df_filtered$Type[grep("^GTEX",sample_type_df_filtered$Case_ID)])
```